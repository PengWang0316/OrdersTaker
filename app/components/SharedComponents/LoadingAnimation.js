import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { LinearProgress } from '@material-ui/core';

const styles = {
  root: {
    height: '100%',
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    minHeight: 150,
    padding: '0 30px'
  },
  div: {
    width: '100%'
  }
};

export const LoadingAnimation = ({ isLoading, classes }) => (
  <Fragment>
    {isLoading && (
      <div className={classes.root}>
        {/* The LinearProgress needs an additional div to wrap it in order to show the position correctly. */}
        <div className={classes.div}>
          <LinearProgress />
        </div>
      </div>
    )}
  </Fragment>
);
LoadingAnimation.propTypes = {
  isLoading: PropTypes.bool,
  classes: PropTypes.object.isRequired
};
LoadingAnimation.defaultProps = { isLoading: true };
/* istanbul ignore next */
const mapStateToProps = state => ({ isLoading: state.isLoading });
export default connect(mapStateToProps, null)(withStyles(styles)(LoadingAnimation));
