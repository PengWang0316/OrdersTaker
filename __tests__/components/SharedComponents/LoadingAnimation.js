import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import { LoadingAnimation } from '../../../app/components/SharedComponents/LoadingAnimation';

jest.mock('@material-ui/core/LinearProgress', () => 'LinearProgress');

describe('LoadingAnimation test', () => {
  const defaultProps = {
    isLoading: true,
    classes: {
      root: 'root',
      div: 'div'
    }
  };
  test('Snapshot with isLoading true', () => expect(renderer.create(<LoadingAnimation {...defaultProps} />).toJSON()).toMatchSnapshot());

  test('Snapshot with isLoading false', () => expect(renderer.create(<LoadingAnimation isLoading={false} />).toJSON()).toMatchSnapshot());
});
