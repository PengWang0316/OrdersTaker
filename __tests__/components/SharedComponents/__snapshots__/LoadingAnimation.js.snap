// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingAnimation test Snapshot with isLoading false 1`] = `null`;

exports[`LoadingAnimation test Snapshot with isLoading true 1`] = `
<div
  className="root"
>
  <div
    className="div"
  >
    <LinearProgress />
  </div>
</div>
`;
